apps/sms/services/js/moz_mobile_message/messaging_serivce.js

```diff
@@ -0,0 +1,187 @@
+/*global BridgeServiceMixin, Settings */
+'use strict';
+
+if (!('BridgeServiceMixin' in self)) {
+  importScripts('/services/js/bridge_service_mixin.js');
+}
+
+(function(exports) {
+  const SERVICE_NAME = 'messaging-service';
+  const CLIENT_NAME = 'mozMobileMessageShim';
+
+  const METHODS = Object.freeze([
+    'sendSMS', 'sendMMS', 'resendMessage'
+  ]);
+
+  // 0 is a valid value so we need to take care to not consider it as a falsy
+  // value. We want to return null for anything that's not a number or a
+  // string containing a number.
+  function sanitizeServiceId(serviceId) {
+    if (serviceId == null || // null or undefined
+        isNaN(+serviceId)) {
+      serviceId = null;
+    } else {
+      serviceId = +serviceId;
+    }
+
+    return serviceId;
+  }
+
+  function getSendOptionsFromServiceId(serviceId) {
+    var sendOpts;
+
+    if (serviceId != null && // not null, not undefined
+        Settings.hasSeveralSim()) {
+      sendOpts = { serviceId: serviceId };
+    }
+
+    return sendOpts;
+  }
+
+  var MessagingService = {
+    /**
+     * Initializes our service using the Service mixin's initService and
+     * requests mozMobileMessage shim as client.
+     */
+    init() {
+      this.initService();
+
+      if (!('bridge' in self)) {
```

Comment:
oh I think you mean to do that for consistency because the client needs the bridge ? then I'd do a separate function "ensureBridge" that would import it, and I'd call it at the start of init().